test/mock/services/signets_api_mock.dart

```diff
@@ -3,37 +3,32 @@ import 'package:mockito/mockito.dart';
 import 'package:notredame/core/managers/course_repository.dart';
 
 // MODELS
-import 'package:notredame/core/models/course_activity.dart';
-import 'package:notredame/core/models/course_evaluation.dart';
-import 'package:notredame/core/models/course_summary.dart';
-import 'package:notredame/core/models/profile_student.dart';
-import 'package:notredame/core/models/program.dart';
-import 'package:notredame/core/models/schedule_activity.dart';
-import 'package:notredame/core/models/session.dart';
-import 'package:notredame/core/models/course.dart';
+import 'package:ets_api_clients/models.dart';
 
 // SERVICE
-import 'package:notredame/core/services/signets_api.dart';
-import 'package:notredame/core/utils/api_exception.dart';
+import 'package:ets_api_clients/clients.dart';
+import 'package:ets_api_clients/exceptions.dart';
 
 /// Mock for the [SignetsApi]
-class SignetsApiMock extends Mock implements SignetsApi {
-  static const signetsException =
-      ApiException(prefix: SignetsApi.tag, message: "");
+class SignetsAPIClientMock extends Mock implements SignetsAPIClient {
```

Comment:
I think it's possible because SharedPreferences works like that but the Mock should be provided by the new packages 😄